### PR TITLE
[iceoryx] Update to 2.0.3

### PR DESCRIPTION
**Pull request by ex-purple**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/iceoryx/portfile.cmake [hunk 1]
```diff
@@ -6,8 +6,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO eclipse-iceoryx/iceoryx
     REF "v${VERSION}"
-    SHA512 a354ed59eab8730238c828a6477bc6d9f8a5c27009e72406dd2f081cd1d490888d6c1ef31609b35599d30da6e32c1d9ddfaad2f5f50360dcd95015febba3d1ff
-    HEAD_REF master
+    SHA512 b860452373dc41d6b0dd0042d1fc4c97621f143d20e9354a1572126ab815ee0cec6548ddfab9fa276943ed33d9ffd8810243e67eb5fae1d11f18ef0b2b074650    HEAD_REF master
```
Comment (jimwang118) [inline ports/iceoryx/portfile.cmake:9]:
```suggestion
    SHA512 b860452373dc41d6b0dd0042d1fc4c97621f143d20e9354a1572126ab815ee0cec6548ddfab9fa276943ed33d9ffd8810243e67eb5fae1d11f18ef0b2b074650    
    HEAD_REF master
```

**Review by jimwang118 — DISMISSED** [review]

**Comment by ex-purple** [comment]
@jimwang118 ping

**Comment by jimwang118** [comment]
> @jimwang118 ping

If you delete the space after SHA256, this PR will be fine, and then it can be merged.

**Comment by ex-purple** [comment]
> If you delete the space after SHA512

@jimwang118 Done

**Review by jimwang118 — APPROVED** [review]

**Pull request merged**